### PR TITLE
Chore: Update to latest Cortex

### PR DESCRIPTION
**Pull request by chaudum**
**What this PR does**:

Upgrades the Cortex dependency to the latest commit.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: cmd/tempo/app/modules.go [hunk 1]
```diff
@@ -248,13 +249,23 @@ func (t *App) initStore() (services.Service, error) {
...
+	reg := prometheus.DefaultRegisterer
+	t.cfg.MemberlistKV.MetricsRegisterer = reg
 	t.cfg.MemberlistKV.MetricsNamespace = metricsNamespace
 	t.cfg.MemberlistKV.Codecs = []codec.Codec{
 		ring.GetCodec(),
 	}
 
-	t.MemberlistKV = memberlist.NewKVInitService(&t.cfg.MemberlistKV, log.Logger)
+	dnsProviderReg := prometheus.WrapRegistererWithPrefix(
+		"cortex_",
```
Comment (joe-elliott) [inline cmd/tempo/app/modules.go:260]: probably don't want to use the `cortex_` provider here. `tempo_`?

**Review by mdisibio — APPROVED** [review]

File: cmd/tempo/app/modules.go (hunk 1 above)
Comment (MichelHollands) [reply cmd/tempo/app/modules.go:260]: Good catch. I'll update that.

**Review by joe-elliott — APPROVED** [review]
looks like you have some conflicts with main, but other than that lgtm

**Comment by MichelHollands** [comment]
> looks like you have some conflicts with main, but other than that lgtm

We can't keep up with you :) It's fixed now.

**Pull request merged**